Test/OutputTestingHtml.py

```diff
@@ -103,29 +113,103 @@ def getImageCompareResultsTable(slnInfo):
         maxCols = 0
         #table needs max num of screenshots plus one columns
         for result in slnInfo.systemResultList:
-            if len(result.CompareResults) > maxCols:
-                maxCols = len(result.CompareResults)
-        html = '<table style="width:100%" border="1">\n'
-        html += '<tr>\n'
-        html += '<th colspan=\'' + str(maxCols + 1) + '\'>Image Compare Tests</th>\n'
-        html += '</tr>\n'
-        html += '<th>Test</th>\n'
-        for i in range (0, maxCols):
-            html += '<th>SS' + str(i) + '</th>\n'
+            if len(result.CompareImageResults) > maxCols:
+                maxCols = len(result.CompareImageResults)
+        
+        if(maxCols > 0):        
+            html = '<table style="width:100%" border="1">\n'
+            html += '<tr>\n'
+            html += '<th colspan=\'' + str(maxCols + 1) + '\'>Image Compare Tests</th>\n'
+            html += '</tr>\n'
+            html += '<th>Test</th>\n'
+            for i in range (0, maxCols):
+                html += '<th>SS' + str(i) + '</th>\n'
+            for result in slnInfo.systemResultList:
+                if len(result.CompareImageResults) > 0:
+                    html += '<tr>\n'
+                    html += '<td>' + result.Name + '</td>\n'
+                    for compare in result.CompareImageResults:
+                        if float(compare) > testingUtil.gDefaultImageCompareMargin or float(compare) < 0:
+                            html += '<td bgcolor="red"><font color="white">' + str(compare) + '</font></td>\n'
+                        else:
+                            html += '<td>' + str(compare) + '</td>\n'
+                    html += '</tr>\n'
+            html += '</table>\n'
+            return html
+        else:
+            return ''
+    else:
+        return ''
+
+
+def getMemoryCompareResultsTable(slnInfo):
+    
+    
+    if slnInfo.systemResultList:
+
+        maxCols = 7
+        hasFrameCheck = False 
+        html = ''
+
         for result in slnInfo.systemResultList:
-            if len(result.CompareResults) > 0:
+            if len(result.CompareMemoryFrameResults) > 0:
+                hasFrameCheck = True
+
+        if hasFrameCheck :
+            html += '<table style="width:100%" border="1">\n'
+            html += '<tr>\n'
+            html += '<th colspan=\'' + str(maxCols) + '\'> Memory Frame Checks </th> \n'
+            html += '</tr>\n'            
```

Comment:
Nitpick, but there are a bunch of lines with a lot of trailing whitespace in this file.